supporting_scripts/analysis-of-endpoint-connections/src/main/java/de/tum/cit/endpointanalysis/RestCallAnalyzer.java

```diff
@@ -0,0 +1,127 @@
+package de.tum.cit.endpointanalysis;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class RestCallAnalyzer {
+
+    private static final String REST_CALL_ANALYSIS_RESULT_PATH = "restCallAnalysisResult.json";
+
+    private static final Logger logger = LoggerFactory.getLogger(RestCallAnalyzer.class);
+
+    public static void main(String[] args) {
+        analyzeRestCalls();
+        printRestCallAnalysisResult();
+    }
+
+    /**
+     * Analyzes REST calls and matches them with endpoints.
+     *
+     * This method reads endpoint and REST call information from JSON files,
+     * compares them to find matching endpoints for each REST call, and writes
+     * the analysis result to a JSON file. REST calls without matching endpoints
+     * are also recorded.
+     */
```

Comment:
I have read the exaxt same comment in `EndpointAnalyzer` - I am a bit confused now 😅 

Why do we have two classes doing the same thing? Could we refine the description to make it more accurate?

Reply:
The two classes are similar but not identical. The `RestCallAnalyzer` analyzes the REST Calls and focuses on them having a matching Endpoint, whereas the `EnpointAnalyzer` focuses on Endpoints and identifying by what endpoints they are used. As there are WildCard endpoints but no wildCard REST calls, I have not found a way to combine the two analyzers into one. I can have a look into that again, and if I come up with a better solution, I will implement it in a follow-up.

Reply:
Yeah, I have seen the differences

I just meant that I think a refinement of the javadoc might be beneficial here 
e.g.

> The RestCallAnalyzer analyzes the client REST Calls and focuses on them having a matching Endpoint on the server

And for the `EnpointAnalyzer` I did not quite get the intend 😅 
We are checking if the endpoints in the server call other endpoints? Why would we do that; we could just call the underlying method instead of calling the REST endpoint?

Reply:
> We are checking if the endpoints in the server call other endpoints? Why would we do that; we could just call the underlying method instead of calling the REST endpoint?

This, indeed, would not be sensible. What we are checking is whether the endpoint is ever called by the Artemis client and where. With this information, it could be possible to analyze and compare the authorization of the client and the server in the future.